test/src/test/java/org/corfudb/integration/CorfuReplicationUpgradeIT.java

```diff
@@ -937,4 +939,46 @@ private void verifyVersion(CorfuStore corfuStore, String expectedVersion, boolea
         Assert.assertEquals(isUpgraded, actualUpgradedFlag);
         log.info("Verified version");
     }
+
+    /**
+     * Code coverage test for the simple LRRollingUpgradeHandler to test if we are able to successfully
+     * 1. Simulate startRollingUpgrade()
+     * 2. verify that rolling upgrade is detected
+     * 3. simulate endRollingUpgrade()
+     * 4. verify that data migration and rolling upgrade end is detected
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testLocalClusterRollingUpgrade() throws Exception {
+        log.info(">> Setup replication for testing during rolling upgrade of active cluster");
+        setupActiveAndStandbyCorfu();
+
+        DefaultAdapterForUpgradeActive defaultAdapterForUpgradeActive = new DefaultAdapterForUpgradeActive(activeRuntime);
+        defaultAdapterForUpgradeActive.startRollingUpgrade(corfuStoreActive);
+        LRRollingUpgradeHandler rollingUpgradeHandler = new LRRollingUpgradeHandler(defaultAdapterForUpgradeActive);
+
+        try (TxnContext txnContext = corfuStoreActive.txn(DefaultAdapterForUpgradeActive.NAMESPACE)) {
+            Assert.assertTrue(rollingUpgradeHandler.isLRUpgradeInProgress(txnContext));
+        }
+
+        defaultAdapterForUpgradeActive.endRollingUpgrade(corfuStoreActive);
+
+        try (TxnContext txnContext = corfuStoreActive.txn(DefaultAdapterForUpgradeActive.NAMESPACE)) {
+            Assert.assertFalse(rollingUpgradeHandler.isLRUpgradeInProgress(txnContext));
+        }
+
+        if (activeCorfu != null) {
```

Comment:
Two minor comments:
(1) I know this is not part of your PR but seems reasonable to have  as a cleanup a cleanup method with the 'After' annotation as this should run for every test. 
(2) Is this properly rebased? I believe Shama had renamed all these to source/sink.

Reply:
(1) Sounds good - i think in the follow up PR where we add the real code in the `migrateData()` paths we will need a proper cleanup or at least a check
(2) Source sink is pushed to `log_replication_v2` branch